Ghidra/Features/Decompiler/src/main/java/ghidra/app/decompiler/DecompileCallback.java

```diff
@@ -1336,13 +1335,20 @@ public StringData getStringData(String addrString, String dtName, String dtId) {
 				}
 				length -= diff;
 				MemoryBufferImpl buf = new MemoryBufferImpl(program.getMemory(), addr, 64);
-				stringInstance = dataType.getStringDataInstance(buf, settings, length);
+				if (data.getDataType() instanceof AbstractStringDataType) {
```

Comment:
Just in case, I don't think the comment above applies here. This should be correct.

Reply:
Sorry if I missed anything, but I don't get what comment don't apply here?